src/routes/jira/jira-get-connections-backfill-status.ts

```diff
@@ -0,0 +1,110 @@
+import {
+	mapSyncStatus,
+	ConnectionSyncStatus,
+	getRetryableFailedSyncErrors
+} from "~/src/util/github-installations-helper";
+
+type SubscriptionBackfillState = {
+	totalRepos?: number;
+	syncedRepos?: number;
+	syncStatus: ConnectionSyncStatus;
+	isSyncComplete: boolean;
+	backfillSince?: string;
+	failedSyncErrors?: Record<string, number>;
+	syncWarning?: string;
+};
+
+type BackFillType = {
+	[key: string]: SubscriptionBackfillState;
+};
+
+export const JiraGetConnectionsBackfillStatus = async (
+	req: Request,
+	res: Response,
+	next: NextFunction
+): Promise<void> => {
+	try {
+		const { jiraHost: localJiraHost } = res.locals;
+		const subscriptionIds = String(req.query?.subscriptionIds)
+			.split(",")
+			.map(Number)
+			.filter(Boolean);
+
+		if (subscriptionIds.length === 0) {
+			req.log.warn("Missing Subscription IDs");
+			res.status(400).send("Missing Subscription IDs");
+			return;
+		}
+
+		const subscriptions = await Subscription.findAllForSubscriptionIds(subscriptionIds);
+
+		const resultSubscriptionIds = subscriptions.map(
+			(subscription) => subscription.id
+		);
+
+		if (!subscriptions || subscriptions.length === 0) {
+			req.log.error("Missing Subscription");
+			res.status(400).send("Missing Subscription");
+			return;
+		}
+
+		const jiraHosts = subscriptions.map(
+			(subscription) => subscription?.jiraHost
+		);
+
+		const jiraHostsMatched = jiraHosts.every(
+			(jiraHost) => jiraHost === localJiraHost
+		);
+
+		if (!jiraHostsMatched) {
+			req.log.error("mismatched Jira Host");
+			res.status(403).send("mismatched Jira Host");
+			return;
+		}
+		const subscriptionsById = groupBy(subscriptions, "id");
+		const backfillStatus = await getBackfillStatus(subscriptionsById);
+		const isBackfillComplete = getBackfillCompletionStatus(backfillStatus);
+		res.status(200).send({
+			data: {
+				subscriptions: backfillStatus,
+				isBackfillComplete,
+				subscriptionIds: resultSubscriptionIds
+			}
+		});
+	} catch (error) {
+		return next(new Error(`Failed to render connected repos`));
+	}
+};
+
+const getBackfillCompletionStatus = (backfillStatus: BackFillType): boolean =>
+	Object.values(backfillStatus).every(
+		(backFill: SubscriptionBackfillState): boolean => backFill?.isSyncComplete
+	);
+
+const getBackfillStatus = async (subscriptionsById): Promise<BackFillType> => {
+	const backfillStatus: BackFillType = {};
+	for (const subscriptionId in subscriptionsById) {
+		const subscription = subscriptionsById[subscriptionId][0];
+		const isSyncComplete =
+			subscription?.syncStatus === SyncStatus.COMPLETE ||
+			subscription?.syncStatus === SyncStatus.FAILED;
+		const failedSyncErrors = await getRetryableFailedSyncErrors(subscription);
+
+		backfillStatus[subscriptionId] = {
+			isSyncComplete,
+			syncStatus: mapSyncStatus(subscription?.syncStatus),
+			totalRepos: subscription?.totalNumberOfRepos,
+			syncedRepos: await RepoSyncState.countFullySyncedReposForSubscription(
```

Comment:
if either of the awaits fail inside the loop it will short circuit so might wanna try catch around it so it keeps going, unless you want it to quit then ignore more lol

```
const failedSyncErrors = await getRetryableFailedSyncErrors(subscription);
syncedRepos: await RepoSyncState.countFullySyncedReposForSubscription(
```

Reply:
well, this is a good suggestion!
Yes in this case till the time we dont get a separate UX to show the error when we fail backfill status polling fails for a particular subscription
it will be better to exclude that result from final result and continue with others